### DISCUSSION TITLE
Laravel 11 DB Transaction and Validation Order Issue

### DISCUSSION THREAD

**Discussion by Abhishek-Jetani** [laravel/framework#52413] [General]
### Laravel Version

11.15.0

### PHP Version

8.2.0

### Database Driver & Version

Mysql & 5.2.0

### Description

In Laravel 10 and previous versions, the following code worked perfectly without any issues. However, in Laravel 11, I encountered a problem when using database transactions with validation.

**Issue:**

When I add `DB::beginTransaction();` above the validation code, the validation messages do not display in the Blade view file.
![not_working_1](https://github.com/user-attachments/assets/d888c8f4-0ff9-4b6f-a1c2-bf8743f1e434)

### Steps To Reproduce

1. Add `DB::beginTransaction();` before the validation logic.

2. Perform validation.

3. If validation fails, the error messages do not show up in the Blade view file.

**Non-working Code:**

**Solutions:**

1. Solution 1:
Adding `DB::rollBack();` above the redirect statement with errors works.
![working_2](https://github.com/user-attachments/assets/1c618431-3f31-4246-bf5e-eec758e9b5a8)

2. Solution 2:
Moving `DB::beginTransaction();` after handling validation errors works perfectly.
![working_1](https://github.com/user-attachments/assets/155c6322-136c-44a9-922e-9705d2839da8)

**Question:**
Is this a bug in Laravel 11, or have there been any changes to the database methods that could cause this issue? This code works without any issues in Laravel 10 and previous versions.

Thank you for your time and assistance.

**Comment by ghost** [comment]
What you just posted is just common sense :)

**Reply by Abhishek-Jetani** [reply]
😏

**Reply by Abhishek-Jetani** [reply]
I would like to determine whether this is a bug or if there have been changes to the database transaction methods in Laravel 11 that could be causing this problem.
